doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+formatted as math.
+
+In general, the rule of thumb is to consider if the variable in question were
+something that rendered differently in code and in math. For example, the Greek
+letter α would be written as ``alpha`` in code and $\alpha$ in LaTeX. The
+reason being that $\alpha$ cannot be used in contexts referring to Python code
+because it is not valid Python, and conversely ``alpha`` would be incorrect in
+a math context because it does not render as the Greek letter (α).
+
+Style Preferences
+-----------------
+
+**Spelling and Punctuation**
+
+All narrative writing in SymPy follows American spelling and punctuation
+standards. For example, “color” is preferred over “colour” and commas should be
+placed inside of quotation marks.
+
+**Examples**
+
+::
+
+  If the ``line_color`` aesthetic is a function of arity 1, then the coloring is
+  a function of the x value of a point.
+
+  The term "unrestricted necklace," or "bracelet," is used to imply an object
+  that can be turned over or a sequence that can be reversed.
+
+If there is any ambiguity about the spelling of a word, such as in the case of
+a function named after a person, refer to the spelling of the actual SymPy
+function.
+
+For example, Chebyshev polynomials are named after Pafnuty Lvovich Tchebychev,
+whose name is sometimes transliterated from Russian to be spelled with a “T,”
+but in SymPy it should always be spelled “Chebyshev” to refer to the SymPy
+function.
+
+**Example**
+
+::
+
+  class chebyshevt(OrthogonalPolynomial):
+      r"""
+      Chebyshev polynomial of the first kind, $T_n(x)$
+      ...
+
+      """
+
+**Capitalization**
+
+Title case capitalization is preferred in all SymPy headings.
+
+**Example**
+
+::
+
+  What is Symbolic Computation?
+  -----------------------------
+
+**Grammar**
+
+WORK IN PROGRESS: Google Season of Docs technical writer will track common
+grammar mistakes (e.g., using certain articles correctly) while editing and add
+to this guide.
+
+Tone Preferences
+----------------
+
+Across SymPy documentation, please write in:
+
+* The present tense (e.g., In the following section, we are going to learn...)
+* The first-person inclusive plural (e.g., We did this the long way, but now we
+  can try it the short way...)
+* Use the generic pronoun “you” instead of “one.” Or use “the reader” or “the
+  user.” (e.g., You can access this function by... The user can then access
+  this function by...)
+* Use the gender-neutral pronoun “they” instead of “he” or “she.” (e.g., A good
+  docstring tells the user exactly what they need to know.)
+
+Avoid extraneous or belittling words such as “obviously,” “easily,” “simply,”
+“just,” or “straightforward.”
+
+Avoid unwelcoming or judgement-based phrases like “That is wrong.” Instead use
+friendly and inclusive language like “A common mistake is...”
+
+Avoid extraneous phrases like, “we just have to do one more thing.”
+
+Docstring Guidelines
+====================
```

Comment:
RST allows lots of different characters to for underling section and subsection headers. And you can control the headling level using any order of these heading symbols. This means that each file can have a different order of the heading underline symbols. We should recommend using a specific order for sympy so it is consistent. For example:

- `==`: `<h1>`
- `--`: `<h2>`
- `~~`: <h3>`

or whatever order and symbols we want to choose.

Reply:
Great, we should copy that!

Reply:
I just did some experimenting and it looks like when you do the overline it doesn't matter which symbol you choose, it still gives you the top level heading (in SymPy's current theme at least). But it's probably still helpful to think of certain symbols being for higher level headings, and this recommendation is very useful so I'll add it to the formatting section. Thanks @mgeier!

Reply:
@lglattly The reST format doesn't have a fixed mapping from symbols to levels.
You as a user can decide that (for each single page). Whatever you use first, becomes the first level, what you use second becomes the second level etc.

The final absolute heading level depends on how deeply nested the `toctree` is.

This is independent of the theme.